src/session/session_api.c

```diff
@@ -2725,8 +2726,12 @@ __open_session(WT_CONNECTION_IMPL *conn, WT_EVENT_HANDLER *event_handler, const
      * Configuration: currently, the configuration for open_session is the same as
      * session.reconfigure, so use that function.
      */
+#ifndef HAVE_CONTROL_POINT
     if (config != NULL)
         WT_ERR(__session_reconfigure((WT_SESSION *)session_ret, config));
+#else
+    WT_ERR(__session_reconfigure((WT_SESSION *)session_ret, config));
+#endif
 
```

Comment:
This change is not needed. Delete new lines 2729, and 2732-2734.

Reply:
The reason why this is needed is because config refers to the configuration string that is passed into session open. It is possible that this is NULL. If the config is NULL, we do not get into the the call `SESSION_API_CALL_PREPARE_NOT_ALLOWED(session, ret, reconfigure, config, cfg);` which populates the configuration with default values.

Reply:
If you keep this you need to make sure that __ session_reconfigure() works when passed config=NULL. I don't think that it does, for example:
__ session_reconfigure() -> __ wt_txn_context_check() -> __ wt_config_getones() -> __ wt_config_init() -> __ wt_config_initn() -> Line 29 conf->session = session; which crashes if conf == NULL.

Why does NULL need to be supported now when it was not before? If NULL really needs to be supported it should be supported here (Only the HAVE_CONTROL_POINT part is shown below):
      if (config != NULL)
          WT_ERR(__session_reconfigure((WT_SESSION *)session_ret, config));
      else
           Do what is needed to support null;

Reply:
as discussed we have fix this now.